BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -484,12 +461,38 @@ - (void)webViewDidFinishLoad:(UIWebView *)webView
     [self.activityIndicator stopAnimating];
 }
 
-#pragma mark - NSURLConnectionDelegate methods
+#pragma mark - NSURLSessionDataTaskDelegate methods
 
-- (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge
+- (void)URLSession:(NSURLSession *)session
+          dataTask:(NSURLSessionDataTask *)dataTask
+didReceiveResponse:(nonnull NSURLResponse *)response
+ completionHandler:(nonnull void (^)(NSURLSessionResponseDisposition))completionHandler
 {
-    BOXLog(@"connection %@ did receive authentication challenge %@", connection, challenge);
+    BOXLog(@"URLSessionDataTask %@ did receive response %@", dataTask, response);
+    if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+        BOXLog(@"HTTP Headers were: %@", [(NSHTTPURLResponse *)response allHeaderFields]);
+    }
+    self.connectionResponse = response;
+    [self.connectionData setLength:0];
```

Comment:
We need to reset it here because on receiving a response a new stream of data might be coming in.